File: Classes/EVEOnlineAPI/EVE/EVEFacWarTopStats.m
//
//  EVEFacWarTopStats.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 8/19/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "EVEFacWarTopStats.h"

@implementation EVEFacWarTopStatsCharactersItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"factionID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"factionName":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"againstID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"againstName":@{@"type":@(EVEXMLSchemePropertyTypeString)}};
	return scheme;
}

@end


@implementation EVEFacWarTopStatsCorporationsItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"factionID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"factionName":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"againstID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"againstName":@{@"type":@(EVEXMLSchemePropertyTypeString)}};
	return scheme;
}


@end


@implementation EVEFacWarTopStatsFactionsItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"factionID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"factionName":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"againstID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"againstName":@{@"type":@(EVEXMLSchemePropertyTypeString)}};
	return scheme;
}

@end

@implementation EVEFacWarTopStatsItem
@end

@implementation EVEFacWarTopStatsCharacters

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"killsYesterday":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsCharactersItem class]},
				   @"killsLastWeek":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsCharactersItem class]},
				   @"killsTotal":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsCharactersItem class]},
				   @"victoryPointsYesterday":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsCharactersItem class]},
				   @"victoryPointsLastWeek":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsCharactersItem class]},
				   @"victoryPointsTotal":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsCharactersItem class]}};
	return scheme;
}

@end

@implementation EVEFacWarTopStatsCorporations

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"killsYesterday":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsCorporationsItem class]},
				   @"killsLastWeek":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsCorporationsItem class]},
				   @"killsTotal":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsCorporationsItem class]},
				   @"victoryPointsYesterday":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsCorporationsItem class]},
				   @"victoryPointsLastWeek":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsCorporationsItem class]},
				   @"victoryPointsTotal":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsCorporationsItem class]}};
	return scheme;
}

@end


@implementation EVEFacWarTopStatsFactions

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"killsYesterday":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsFactionsItem class]},
				   @"killsLastWeek":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsFactionsItem class]},
				   @"killsTotal":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsFactionsItem class]},
				   @"victoryPointsYesterday":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsFactionsItem class]},
				   @"victoryPointsLastWeek":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsFactionsItem class]},
				   @"victoryPointsTotal":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarTopStatsFactionsItem class]}};
	return scheme;
}

@end

@implementation EVEFacWarTopStats

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"characters":@{@"type":@(EVEXMLSchemePropertyTypeObject), @"class":[EVEFacWarTopStatsCharacters class]},
				   @"corporations":@{@"type":@(EVEXMLSchemePropertyTypeObject), @"class":[EVEFacWarTopStatsCorporations class]},
				   @"factions":@{@"type":@(EVEXMLSchemePropertyTypeObject), @"class":[EVEFacWarTopStatsFactions class]}};
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVEFacWarTopStats.swift
//
//  EVEFacWarTopStats.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 30.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit


public class EVEFacWarTopStatsCharactersItem: EVEObject {
	public var characterID: Int = 0
	public var characterName: String = ""
	public var kills: Int = 0
	
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"characterID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"characterName":EVESchemeElementType.String(elementName:nil, transformer:nil),
			"kills":EVESchemeElementType.Int(elementName:nil, transformer:nil),
		]
	}
}

public class EVEFacWarTopStatsCorporationsItem: EVEObject {
	public var corporationID: Int = 0
	public var corporationName: String = ""
	public var kills: Int = 0
	
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"corporationID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"corporationName":EVESchemeElementType.String(elementName:nil, transformer:nil),
			"kills":EVESchemeElementType.Int(elementName:nil, transformer:nil),
		]
	}
}

public class EVEFacWarTopStatsFactionsItem: EVEObject {
	public var factionID: Int = 0
	public var factionName: String = ""
	public var kills: Int = 0
	
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"factionID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"factionName":EVESchemeElementType.String(elementName:nil, transformer:nil),
			"kills":EVESchemeElementType.Int(elementName:nil, transformer:nil),
		]
	}
}

public class EVEFacWarTopStatsCharacters: EVEObject {
	public var killsYesterday: [EVEFacWarTopStatsCharactersItem] = []
	public var killsLastWeek: [EVEFacWarTopStatsCharactersItem] = []
	public var killsTotal: [EVEFacWarTopStatsCharactersItem] = []
	public var victoryPointsYesterday: [EVEFacWarTopStatsCharactersItem] = []
	public var victoryPointsLastWeek: [EVEFacWarTopStatsCharactersItem] = []
	public var victoryPointsTotal: [EVEFacWarTopStatsCharactersItem] = []
	
	public override init() {
		super.init()
	}
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"killsLastWeek":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsCharactersItem.self, transformer: nil),
			"killsTotal":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsCharactersItem.self, transformer: nil),
			"victoryPointsYesterday":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsCharactersItem.self, transformer: nil),
			"victoryPointsLastWeek":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsCharactersItem.self, transformer: nil),
			"victoryPointsTotal":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsCharactersItem.self, transformer: nil),
		]
	}
}

public class EVEFacWarTopStatsCorporations: EVEObject {
	public var killsYesterday: [EVEFacWarTopStatsCorporationsItem] = []
	public var killsLastWeek: [EVEFacWarTopStatsCorporationsItem] = []
	public var killsTotal: [EVEFacWarTopStatsCorporationsItem] = []
	public var victoryPointsYesterday: [EVEFacWarTopStatsCorporationsItem] = []
	public var victoryPointsLastWeek: [EVEFacWarTopStatsCorporationsItem] = []
	public var victoryPointsTotal: [EVEFacWarTopStatsCorporationsItem] = []
	
	public override init() {
		super.init()
	}
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"killsLastWeek":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsCorporationsItem.self, transformer: nil),
			"killsTotal":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsCorporationsItem.self, transformer: nil),
			"victoryPointsYesterday":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsCorporationsItem.self, transformer: nil),
			"victoryPointsLastWeek":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsCorporationsItem.self, transformer: nil),
			"victoryPointsTotal":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsCorporationsItem.self, transformer: nil),
		]
	}
}

public class EVEFacWarTopStatsFactions: EVEObject {
	public var killsYesterday: [EVEFacWarTopStatsFactionsItem] = []
	public var killsLastWeek: [EVEFacWarTopStatsFactionsItem] = []
	public var killsTotal: [EVEFacWarTopStatsFactionsItem] = []
	public var victoryPointsYesterday: [EVEFacWarTopStatsFactionsItem] = []
	public var victoryPointsLastWeek: [EVEFacWarTopStatsFactionsItem] = []
	public var victoryPointsTotal: [EVEFacWarTopStatsFactionsItem] = []
	
	public override init() {
		super.init()
	}
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"killsLastWeek":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsFactionsItem.self, transformer: nil),
			"killsTotal":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsFactionsItem.self, transformer: nil),
			"victoryPointsYesterday":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsFactionsItem.self, transformer: nil),
			"victoryPointsLastWeek":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsFactionsItem.self, transformer: nil),
			"victoryPointsTotal":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsFactionsItem.self, transformer: nil),
		]
	}
}

public class EVEFacWarTopStats: EVEResult {
	public var characters: EVEFacWarTopStatsCharacters = EVEFacWarTopStatsCharacters()
	public var corporations: EVEFacWarTopStatsCorporations = EVEFacWarTopStatsCorporations()
	public var factions: EVEFacWarTopStatsFactions = EVEFacWarTopStatsFactions()
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"characters":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsCharacters.self, transformer: nil),
			"corporations":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsCorporations.self, transformer: nil),
			"factions":EVESchemeElementType.Object(elementName: nil, type: EVEFacWarTopStatsFactions.self, transformer: nil)
		]
	}
}

